src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java

```diff
@@ -872,6 +880,64 @@ public void testYouCantWriteIntoAnExistingDirectory(){
         writeToGenomicsDB(LOCAL_GVCFS, INTERVAL, workspace, 0, false, 0, 1);
     }
 
+    @Test(expectedExceptions = CommandLineException.class)
+    public void testOverwriteWorkspaceAndIncrementalImportCannotBothBeTrue() {
+        final String workspace = createTempDir("genomicsdb-incremental-tests").getAbsolutePath() + "/workspace";
+        writeToGenomicsDB(LOCAL_GVCFS, INTERVAL, workspace, 0, false, 0, 1, false, true, true);
+    }
+
+    @Test(expectedExceptions = UserException.class)
+    public void testIncrementalMustHaveExistingWorkspace() {
+        final String workspace = createTempDir("genomicsdb-incremental-tests").getAbsolutePath();
+        writeToGenomicsDB(LOCAL_GVCFS, INTERVAL, workspace + "workspace2", 0, false, 0, 1, false, false, true);
+    }
+
+    @Test
+    public void testGenomicsDBBasicIncremental() throws IOException {
+        final String workspace = createTempDir("genomicsdb-incremental-tests").getAbsolutePath() + "/workspace";
+        writeToGenomicsDB(LOCAL_GVCFS.subList(0,2), INTERVAL, workspace, 0, false, 0, 1, false, false, false);
+        checkJSONFilesAreWritten(workspace);
+        writeToGenomicsDB(LOCAL_GVCFS.subList(2,3), INTERVAL, workspace, 0, false, 0, 1, false, false, true);
+        checkJSONFilesAreWritten(workspace);
+
+        checkGenomicsDBAgainstExpected(workspace, INTERVAL, COMBINED_WITH_GENOTYPES, b38_reference_20_21, true, ATTRIBUTES_TO_IGNORE, true, false);
+    }
+
+    @Test
+    public void testGenomicsDBIncrementalAndBatchSize1() throws IOException {
+        final String workspace = createTempDir("genomicsdb-incremental-tests").getAbsolutePath() + "/workspace";
+        writeToGenomicsDB(LOCAL_GVCFS.subList(0,2), INTERVAL, workspace, 1, false, 0, 1, false, false, false);
+        checkJSONFilesAreWritten(workspace);
+        writeToGenomicsDB(LOCAL_GVCFS.subList(2,3), INTERVAL, workspace, 1, false, 0, 1, false, false, true);
+        checkJSONFilesAreWritten(workspace);
+
+        checkGenomicsDBAgainstExpected(workspace, INTERVAL, COMBINED_WITH_GENOTYPES, b38_reference_20_21, true, ATTRIBUTES_TO_IGNORE, true, false);
+    }
+
+    @Test
+    public void testGenomicsDBIncrementalAndBatchSize2() throws IOException {
+        final String workspace = createTempDir("genomicsdb-incremental-tests").getAbsolutePath() + "/workspace";
+        writeToGenomicsDB(LOCAL_GVCFS.subList(0,2), INTERVAL, workspace, 2, false, 0, 1, false, false, false);
+        checkJSONFilesAreWritten(workspace);
+        writeToGenomicsDB(LOCAL_GVCFS.subList(2,3), INTERVAL, workspace, 2, false, 0, 1, false, false, true);
+        checkJSONFilesAreWritten(workspace);
+
+        checkGenomicsDBAgainstExpected(workspace, INTERVAL, COMBINED_WITH_GENOTYPES, b38_reference_20_21, true, ATTRIBUTES_TO_IGNORE, true, false);
+    }
+
+    @Test
+    public void testGenomicsDBMultipleIncrementalImports() throws IOException {
```

Comment:
Same here -- instead of expected, compare with one batch of three or the CombineGVCFs output.